src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/FullBlockSectionProcessor.scala

```diff
@@ -70,19 +70,22 @@ trait FullBlockSectionProcessor extends BlockSectionProcessor with FullBlockProc
     */
   object PayloadValidator extends ModifierValidator with ScorexEncoding {
 
-    def validate(m: ErgoPersistentModifier, header: Header, minimalHeight: Int): ValidationResult[Unit] = {
+    def validate(m: BlockSection, header: Header, minimalHeight: Int): ValidationResult[Unit] = {
       failFast
-        .validate(!historyStorage.contains(m.id)) {
-          fatal(s"Modifier ${m.encodedId} is already in history")
-        }
         .validate(header.isCorrespondingModifier(m)) {
           fatal(s"Modifier ${m.encodedId} does not corresponds to header ${header.encodedId}")
         }
         .validateSemantics(isSemanticallyValid(header.id)) {
           fatal(s"Header ${header.encodedId} for modifier ${m.encodedId} is semantically invalid")
         }
-        .validate(header.height >= minimalHeight) {
-          fatal(s"Too old modifier ${m.encodedId}: ${header.height} < $minimalHeight")
+        .validate(!isHeadersChainSynced || header.height >= minimalHeight) {
+          error(s"Too old modifier ${m.encodedId}: ${header.height} < $minimalHeight")
+        }
+        .validate(isHeadersChainSynced) {
+          error("Headers chain is not synced yet")
+        }
```

Comment:
From the two conditions above, validation will always fail.

Reply:
It does not fail, if `isHeadersChainSynced == true && header.height >= minimalHeight`. Rewrote to this single condition